torch/testing/_internal/common_utils.py

```diff
@@ -1358,18 +1358,30 @@ def __exit__(self, exec_type, exec_value, traceback):
         torch.cuda.empty_cache()
 
         for i in range(num_devices):
-            caching_allocator_mem_allocated = torch.cuda.memory_allocated(i)
-            bytes_free, bytes_total = torch.cuda.mem_get_info(i)
-            driver_mem_allocated = bytes_total - bytes_free
 
-            caching_allocator_discrepancy = False
-            driver_discrepancy = False
+            discrepancy_detected = True
 
-            if caching_allocator_mem_allocated > self.caching_allocator_befores[i]:
-                caching_allocator_discrepancy = True
+            # Query memory multiple tiems to ensure leak was not transient
+            for n in range(3):
+                caching_allocator_mem_allocated = torch.cuda.memory_allocated(i)
+                bytes_free, bytes_total = torch.cuda.mem_get_info(i)
+                driver_mem_allocated = bytes_total - bytes_free
+
+                caching_allocator_discrepancy = False
+                driver_discrepancy = False
 
-            if driver_mem_allocated > self.driver_befores[i]:
-                driver_discrepancy = True
+                if caching_allocator_mem_allocated > self.caching_allocator_befores[i]:
+                    caching_allocator_discrepancy = True
+
+                if driver_mem_allocated > self.driver_befores[i]:
+                    driver_discrepancy = True
+
+                if not( caching_allocator_discrepancy or driver_discrepancy):
+                    # Leak was false positive
+                    discrepancy_detected = False
```

Comment:
If `discrepancy_detected` is `False`, we should `break` here, so as not to unnecessarily execute more iterations than necessary. Otherwise, this change unduly triples the number of times we query the caching allocator and mem info on both CUDA and ROCm, even if not necessary, which will probably make it not upstreamable.
cc @jeffdaily

Reply:
ah, good catch @jithunnair-amd. This hole in the logic was a relic of when I was experimenting with a while loop that depended on `discrepancy_detected`. Updating in upcoming commit